### PR TITLE
added feature and test for zipcode validation

### LINKED ISSUE #24: Fix: Do not search for wilaya by zip code if it's invalid
In Algeria all ZIP codes are numeric in a form of `WILAYA_CODE_XXXX`. For example Oran city's code is 31, and it provinces have codes like `31000`, `310094` ...

In this line :

https://github.com/dzcode-io/leblad/blob/5837b9e613b414acf84d4bc8f4b9d0b567f84dbc/src/api/getWilayaByZipCode.js#L18

We are directly looking for the given zip code even if it's not a valid number. I suggest at least checking if the parameter is a number before searching.

### PR DESCRIPTION
**Pull request by xxKeefer**
# Description

Added feature and test for validating zip codes to utils

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I checked that there's no dataset update (can be done by running `npm run update-dataset`)
- [X] `npm test` passes on my machine
- [X] `npm run lint` passes on my machine

### CODE REVIEW

File: src/utils/isValidZipCode.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const ZIP_COUNT = 48073;
+
+/**
+ * Check if a given integer is a valid zip code (between 1000 and 48073)
+ * @param {Number} code zip code
+ * @returns {Boolean}
+ */
+const isValidZipCode = (code) =>
+  Number.isInteger(code) && code >= 1000 && code <= ZIP_COUNT;
```
Comment (Fcmam5) [inline src/utils/isValidZipCode.js]: Your point in #24 is valid IMO, we can pass a string like `'01027'`, so I'd suggest either just checking if code is not `NaN` (with [Number.isNaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN)), or you use `parseInt(code)` then you check for `parsedCode >= 1000 && parsedCode <= ZIP_COUNT;`

**Comment by InternetRamen** [comment]
What should I do? I'm new to open source.

**Comment by Fcmam5** [comment]
> What should I do? I'm new to open source.

You can help us here by reviewing the code, and suggest any improvements. But first, we should wait for the PR author to publish the latest changes and then when the review is requested we can check it.
Or sometimes we notice something that we can improve and it could be outside of this PR scope, you can resolve it in a separate PR.

But for now, everything is fine :) just check our latest issue, it should be a nice welcoming-challenge for your open-source journey 🙂 !

**Comment by xxKeefer** [comment]
# Description
Added feature and test for validating zip codes to utils
function now handles valid strings that can be parsed into into a valid integer, updated test to reflect this.

Fixes #26

# Type of change
Please delete options that are not relevant.

[ X] New feature (non-breaking change which adds functionality)

# Checklist:
[ X] I checked that there's no dataset update (can be done by running npm run update-dataset)
[ X] npm test passes on my machine
[X ] npm run lint passes on my machine

File: src/utils/isValidZipCode.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Check if a given integer is a valid zip code (between 1000 and 48073)
+ * @param {Number} code zip code
+ * @returns {Boolean}
+ */
+const isValidZipCode = (code) => {
+  const parsedCode = parseInt(code, 10);
+  return (
+    !Number.isNaN(parsedCode) &&
```
Comment (Fcmam5) [inline src/utils/isValidZipCode.js]: We can remove this, as if the number is `NaN` our `Number.isInteger(parsedCode)` will return false
```suggestion
```

File: src/utils/isValidZipCode.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const ZIP_COUNT = 48073;
+
+/**
+ * Check if a given integer is a valid zip code (between 1000 and 48073)
+ * @param {Number} code zip code
```
Comment (Fcmam5) [inline src/utils/isValidZipCode.js]: This function still work if we pass '01000' 

```suggestion
 * Check if a given parameter is a valid zip code (between 1000 and 48073)
 * @param {Number|String} code zip code
```

**Review by Fcmam5 — CHANGES_REQUESTED** [review]
Clean! Thank you very much 😄 

I only have the follow suggestions. 
+ please export this file in `index.js` and document it in `README.md`

**Comment by xxKeefer** [comment]
OK, should be good now, can you please look over the documentation in the README, i tried to follow the format but i am not sure if i matched it correctly.

**Review by Fcmam5 — APPROVED** [review]
Awesome! Thank you very much @xxKeefer for your contribution.

**Comment by Fcmam5** [comment]
@all-contributors add @xxKeefer for code

**Pull request merged**